pkg/reconciler/inmemorychannel/dispatcher/inmemorychannel.go

```diff
@@ -86,7 +96,41 @@ func (r *Reconciler) FinalizeKind(ctx context.Context, imc *v1.InMemoryChannel)
 		}
 	}
 	return nil
+}
+
+func (r *Reconciler) patchSubscriberStatus(ctx context.Context, imc *v1.InMemoryChannel) error {
+	after := imc.DeepCopy()
 
+	after.Status.Subscribers = make([]eventingduckv1.SubscriberStatus, 0)
+	for _, sub := range imc.Spec.Subscribers {
+		after.Status.Subscribers = append(after.Status.Subscribers, eventingduckv1.SubscriberStatus{
+			UID:                sub.UID,
+			ObservedGeneration: sub.Generation,
+			Ready:              corev1.ConditionTrue,
+		})
+	}
+	jsonPatch, err := duck.CreatePatch(imc, after)
+	if err != nil {
+		logging.FromContext(ctx).Warnw("Failed to create JSON patch", zap.Error(err))
+		return err
```

Comment:
I see all errors in this PR are explicitly logged before being returned. Is there any specific reason for not wrapping the error with `fmt.Errorf("creating JSON patch: %w", err)` and letting the caller log it?

Reply:
nope, good idea :)